src/Config/Validator/Deploy/EnvironmentVariables.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\App\Error;
+use Magento\MagentoCloud\Config\Environment;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\ValidatorException;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Magento\MagentoCloud\Package\MagentoVersion;
+
+/**
+ * Validates that environment variables doesn't contain redundant variables.
+ * - CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL must be absent in env variables for Magento >= 2.4.0
+ */
+class EnvironmentVariables implements ValidatorInterface
+{
+    /**
+     * @var Environment
+     */
+    private $environment;
+
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var MagentoVersion
+     */
+    private $magentoVersion;
+
+    /**
+     * @param Environment $environment
+     * @param MagentoVersion $magentoVersion
+     * @param Validator\ResultFactory $resultFactory
+     */
+    public function __construct(
+        Environment $environment,
+        MagentoVersion $magentoVersion,
+        Validator\ResultFactory $resultFactory
+    ) {
+        $this->environment = $environment;
+        $this->magentoVersion = $magentoVersion;
+        $this->resultFactory = $resultFactory;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        try {
+            if ($this->magentoVersion->satisfies('>= 2.4.0')
+                && $this->environment->getEnv('CONFIG__STORES__DEFAULT__PAYMENT__BRAINTREE__CHANNEL')
```

Comment:
as you check not only cloud default variable, but also configurations customer sets manually, then you probably need to check existence of this configuration in other sources, isn't it?

Reply:
Under other sources you mean DB? As I understand error happens only if the variable for not existence module is configured through ENV variables.

Reply:
Oh, then OK.